.github/workflows/linter-testing-lib.yaml

```diff
@@ -28,7 +28,7 @@ on:
 
 env:
   RUST_FMT: nightly-2023-04-01-x86_64-unknown-linux-gnu
-  RUST_CLIPPY: 1.66
+  RUST_CLIPPY: 1.67
```

Comment:
Why this?

This should match the minimal supported version stated in Cargo.toml.

Reply:
The CI pipeline failed with version 1.66
https://github.com/Concordium/concordium-rust-smart-contracts/actions/runs/6781383933